.zuul.d/config.yaml

```diff
@@ -3,47 +3,137 @@
-        - scs-daily-security-scan
-    periodic-weekly:
+        - scs-baseline-security-scan-test
+        - scs-full-security-scan-test
+        - scs-greenbone-security-scan-test
+    gate:
       jobs:
-        - scs-weekly-security-scan
-    post:
+        - scs-baseline-security-scan-test
+        - scs-full-security-scan-test
+        - scs-greenbone-security-scan-test
+    periodic-daily:
       jobs:
-        - noop
-    check:
+        - scs-baseline-security-scan
+    periodic-weekly:
       jobs:
-        - noop
+        - scs-full-security-scan
+        - scs-greenbone-security-scan
 
 - job:
-    name: scs-daily-security-scan
+    name: scs-security-scan-base
     parent: base
-    timeout: 7200 #2h
+    description: |
+      Base for any security scanning job
+    timeout: 3600 #1h
+    pre-run: playbooks/pre.yaml
+    post-run: playbooks/post.yaml
+    vars:
+      # Where to store intermediate data
+      scan_work_dir: "{{ ansible_user_dir }}/wrk"
+      # Where to flush (and read) results from
+      scan_results_dir: "{{ ansible_user_dir }}/wrk/scan_results"
+      # Default list of scan targets.
+      # NOTE(gtema): It would be nice in tests to use test instance itself, but
+      # it only exposes SSH port what makes scanning very fast and reports
+      # empty. We potentially want to see how the reports are looking like also 
+      # in tests.
+      scan_targets:
+        - 213.131.230.161
+
+- job:
+    name: scs-baseline-security-scan-base
+    parent: scs-security-scan-base
+    description: |
+      Base for the security scan involving naabu/nuclei/httpx/OWASP Zap.
+
+      OWASP Zap test is performed in the baseline configuration
+    run: playbooks/daily-scan.yaml
+    vars:
+      zap_script: "zap-baseline.py"
+
+- job:
+    name: scs-baseline-security-scan
+    parent: scs-baseline-security-scan-base
+    description: |
+      Perform baseline scan of the resources
     secrets:
      - name: pipeline_conf
        secret: SECRET_SECURITY_INFRA_SCAN_PIPELINE
-    pre-run: playbooks/pre.yaml
-    run: playbooks/daily-scan.yaml
     vars:
       # TODO: engagement_id is not a secret, define id here directly
       # Since in Zuul it is not possible to define vars from secrets on the
       # job level pass a key name down to the job
       engagement: "daily_scan_engagement_id"
-      zap_script: "zap-baseline.py"
 
 - job:
-    name: scs-weekly-security-scan
-    parent: base
+    name: scs-baseline-security-scan-test
+    parent: scs-baseline-security-scan-base
+    description: |
+      Test baseline-security-scan job on the test resources.
+
+      Since this is an untrusted job it does not have access to the secrets
+      and thus not posting results to the DefectDojo.
+    vars:
+      pipeline_conf:
+        dojo_url: https://demo.defectdojo.com/api/v2/import-scan/
+        daily_scan_engagement_id: 1
+        weekly_scan_engagement_id: 1
+        dojo_auth: "Basic YWRtaW46MURlZmVjdGRvam9AZGVtbyNhcHBzZWM="
```

Comment:
yes, demo.defectdojo is new. Idea is to verify artifacts upload. We do not want to use production credentials, since in the test jobs (untrusted) there is no way to use credentials (from security pov). So there are 2 options:
1) use instance for which credentials are not secrets
2) deploy dummy defectdojo as part of the test and upload results there (or mock webserver)

This part of the test can be skipped by not setting the auth. In this case upload is not tried, but we also do not know if it works before we try.